core/jvm/src/main/scala/zio/internal/ZScheduler.scala

```diff
@@ -442,6 +416,36 @@ private final class ZScheduler(autoBlocking: Boolean) extends Executor {
           }
         }
       }
+
+      def markAsBlocking(workerIdx: Int, ignoreIfAutoBlocking: Boolean): Unit =
+        if (blocking || (ignoreIfAutoBlocking && autoBlocking)) ()
+        else {
+          blocking = true
+          val workerId =
+            if (workerIdx >= 0) workerIdx
+            else {
+              val i = workers.indexOf(self)
+              if (i >= 0) i else throw new Error("Catastrophic error: worker not found in workers array")
```

Comment:
what happens if a fiber calls `markCurrentWorkerAsBlocking` multiple times? I think this isn't really a failure

Reply:
The worker will already have `blocking = true`, and the method will exit before we get here

Reply:
See L421

Reply:
However I'm contemplating to wrap this method in a synchronized block just to make it 1000% certain that we somehow another thread doesn't try to run this method at the exact same time. It should never happen, at least not with the current code, but we can't guarantee that some new code addition won't try to do that

Reply:
I'd make it private[zio] or even private[internal], seems like it's only invoked by the fiber itself which should be good enough to defend this case.

btw reviewing a pr in github takes away some of the context, I completely missed the fact `blocking` is a member variable of `Worker` ...

Reply:
The Worker class which the method exists on is private in the ZScheduler. So it can't be invoked outside of ZScheduler

Reply:
PS: I ended up removing the error throwing afterall. While we should never get to that point, it's still not a nice thing to have in the code